tests/unit/nupic/research/temporal_memory_test.py

```diff
@@ -59,368 +72,848 @@ def testInitInvalidParams(self):
-    tm = self.tm
-
-    # No previous predictive cells, no active columns
-    prevPredictiveCells = set()
-    activeColumns      = set()
-    prevMatchingCells = set()
-
-    (activeCells,
-    winnerCells,
-    predictedColumns,
-    predictedInactiveCells) = tm.activateCorrectlyPredictiveCells(prevPredictiveCells,
-                                                                  prevMatchingCells,
-                                                                  activeColumns)
-
-    self.assertEqual(activeCells,      set())
-    self.assertEqual(winnerCells,      set())
-    self.assertEqual(predictedColumns, set())
-    self.assertEqual(predictedInactiveCells, set())
-
-    # No previous predictive cells, with active columns
-
-    prevPredictiveCells = set()
-    activeColumns = set([32, 47, 823])
-    prevMatchingCells = set()
-
-    (activeCells,
-    winnerCells,
-    predictedColumns,
-    predictedInactiveCells) = tm.activateCorrectlyPredictiveCells(prevPredictiveCells,
-                                                                  prevMatchingCells,
-                                                                  activeColumns)
-
-    self.assertEqual(activeCells,      set())
-    self.assertEqual(winnerCells,      set())
-    self.assertEqual(predictedColumns, set())
-    self.assertEqual(predictedInactiveCells, set())
-
-    # No active columns, with previously predictive cells
-
-    prevPredictiveCells = set([0, 237, 1026, 26337, 26339, 55536])
-    activeColumns = set()
-    prevMatchingCells = set()
-
-    (activeCells,
-    winnerCells,
-    predictedColumns,
-    predictedInactiveCells) = tm.activateCorrectlyPredictiveCells(prevPredictiveCells,
-                                                                  prevMatchingCells,
-                                                                  activeColumns)
-
-    self.assertEqual(activeCells,      set())
-    self.assertEqual(winnerCells,      set())
-    self.assertEqual(predictedColumns, set())
-    self.assertEqual(predictedInactiveCells, set())
-
-  def testActivateCorrectlyPredictiveCellsOrphan(self):
-    tm = self.tm
-    tm.predictedSegmentDecrement = 0.001
-    prevPredictiveCells = set([])
-    activeColumns = set([32, 47, 823])
-    prevMatchingCells = set([32, 47])
-
-    (activeCells,
-    winnerCells,
-    predictedColumns,
-    predictedInactiveCells) = tm.activateCorrectlyPredictiveCells(prevPredictiveCells,
-                                                                  prevMatchingCells,
-                                                                  activeColumns)
-
-    self.assertEqual(activeCells, set([]))
-    self.assertEqual(winnerCells, set([]))
-    self.assertEqual(predictedColumns, set([]))
-    self.assertEqual(predictedInactiveCells, set([32,47]))
-
-  def testBurstColumns(self):
+  def testActivateCorrectlyPredictedCells(self):
     tm = TemporalMemory(
+      columnDimensions=[32],
```

Comment:
Use a tuple here. And avoid mutable data types as default values since the same mutable value will be shared for all function calls.